stan/math/prim/err/elementwise_check.hpp

```diff
@@ -124,6 +124,13 @@ inline void elementwise_check(const F& is_good, const char* function,
     }();
   }
 }
+template <typename F, typename T, typename... Indexings,
+          require_stan_closure_t<T>* = nullptr>
+inline void elementwise_check(const F& is_good, const char* function,
+                              const char* name, const T& x, const char* must_be,
+                              const Indexings&... indexings) {
+  // XXX skip closures
```

Comment:
Actually, thinking more, I don't think there's a big advantage to implementing this.

We could expose the variables captured by a closure to checks, but the Math checks wouldn't know in what order its getting them, and then depending on which function was accepting closures it would need to decide which checks to do on which inputs.

I think instead in the ODE solves we check only the arguments passed in explicitly (which this is effectively doing) or we get rid of the infinity checks on the inputs to the ODE solves. I'll make an issue and see if getting rid of the checks altogether is an option.